teps/XXXX-step-workspaces.md

```diff
@@ -0,0 +1,254 @@
+---
+title: step-and-sidecar-workspaces
+authors:
+  - "@sbwsg"
+creation-date: 2020-10-02
+last-updated: 2020-10-02
+status: proposed
+---
+
+-->
+# TEP-XXXX: Step and Sidecar Workspaces
+
+<!--
+Ensure the TOC is wrapped with
+  <code>&lt;!-- toc --&rt;&lt;!-- /toc --&rt;</code>
+tags, and then generate with `hack/update-toc.sh`.
+-->
+
+<!-- toc -->
+- [Summary](#summary)
+- [Motivation](#motivation)
+  - [Goals](#goals)
+  - [Non-Goals](#non-goals)
+- [Requirements](#requirements)
+- [Proposal](#proposal)
+  - [Add <code>workspaces</code> to <code>Steps</code>](#add--to-)
+  - [Add <code>workspaces</code> to <code>Sidecars</code>](#add--to--1)
+  - [User Stories](#user-stories)
+    - [Story 1](#story-1)
+    - [Story 2](#story-2)
+    - [Story 3](#story-3)
+    - [Story 4](#story-4)
+  - [Risks and Mitigations](#risks-and-mitigations)
+- [Design Details](#design-details)
+- [Drawbacks](#drawbacks)
+- [Alternatives](#alternatives)
+  - [Use an explicit volumeMount instead](#use-an-explicit-volumemount-instead)
+    - [Advantages](#advantages)
+    - [Drawbacks](#drawbacks-1)
+- [Upgrade &amp; Migration Strategy (optional)](#upgrade--migration-strategy-optional)
+- [References (optional)](#references-optional)
+<!-- /toc -->
+
+## Summary
+
+## Motivation
+
+### Goals
+
+- Provide a mechanism to limit the exposure of sensitive workspaces to only those Steps in a Task
+that actually require access to them.
+- Provide explicit access to Task workspaces from sidecars without using `volumeMounts`.
+
+### Non-Goals
+
+- Validation of the contents of a WorkspaceBinding
+- Validation of the configuration of a WorkspaceBinding
+- Overriding configuration of a WorkspaceBinding
```

Comment:
apologies for this TOTALY nitpick, but is it possible to call "WorkspaceBinding" something else here, or define it? just cuz you'd have to be familiar with the code to know what this is, I assume this is:

https://github.com/tektoncd/pipeline/blob/ae69af262f4601011481fe2e6cb28f08a4f16003/pkg/apis/pipeline/v1beta1/workspace_types.go#L50-L77

i.e. where you specify which workspace to actually use at runtime (https://github.com/tektoncd/pipeline/blob/master/docs/taskruns.md#specifying-workspaces)?

I'm a bit confused about why these are non-goals - maybe it's because it seems like this proposal doesn't involve changing anything about the runtime configuration, if I understand correctly? (cuz at runtime you still specify workspaces in the same way you did before?)